app/src/main/java/com/mapbox/navigation/examples/basics/MultipleWayPointsActivity.kt

```diff
@@ -0,0 +1,556 @@
+            Style.MAPBOX_STREETS
+        ) {
+            // add long click listener that search for a route to the clicked destination
+            binding.mapView.gestures.addOnMapLongClickListener { point ->
+                addWaypoint(point)
+                true
+            }
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        mapboxNavigation.registerRoutesObserver(routesObserver)
+        mapboxNavigation.registerLocationObserver(locationObserver)
+        mapboxNavigation.registerRouteProgressObserver(routeProgressObserver)
+        mapboxNavigation.registerVoiceInstructionsObserver(voiceInstructionsObserver)
+
+        if (mapboxNavigation.getRoutes().isEmpty()) {
+            // if simulation is enabled (ReplayLocationEngine set to NavigationOptions)
+            // but we're not simulating yet,
+            // push a single location sample to establish origin
+            mapboxReplayer.pushEvents(
+                listOf(
+                    ReplayRouteMapper.mapToUpdateLocation(
+                        eventTimestamp = 0.0,
+                        point = Point.fromLngLat(-122.39726512303575, 37.785128345296805)
+                    )
+                )
+            )
+            mapboxReplayer.playFirstLocation()
+        }
+    }
+
+    override fun onBackPressed() {
+        // clear current route
+        if (mapboxNavigation.getRoutes().isNotEmpty()) {
+            stopSimulation()
+            mapboxNavigation.setRoutes(emptyList())
+            addedWaypoints.clear()
+        } else {
+            super.onBackPressed()
+        }
+    }
+
+    override fun onStop() {
+        super.onStop()
+        // unregister event listeners to prevent leaks or unnecessary resource consumption
+        mapboxNavigation.unregisterRoutesObserver(routesObserver)
+        mapboxNavigation.unregisterLocationObserver(locationObserver)
+        mapboxNavigation.unregisterRouteProgressObserver(routeProgressObserver)
+        mapboxNavigation.unregisterVoiceInstructionsObserver(voiceInstructionsObserver)
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        MapboxNavigationProvider.destroy()
+        speechApi.cancel()
+        voiceInstructionsPlayer.shutdown()
+    }
+
+    private fun addWaypoint(destination: Point) {
+        val originLocation = navigationLocationProvider.lastLocation
+        val originPoint = originLocation?.let {
+            Point.fromLngLat(it.longitude, it.latitude)
+        } ?: return
+
+        if (addedWaypoints.isEmpty) {
+            addedWaypoints.addRegular(originPoint)
+        }
+
+        when (val index = binding.waypointTypeView.selectedTabPosition) {
+            1 -> addedWaypoints.addNamed(destination, "Example of the named waypoint")
+            0 -> addedWaypoints.addRegular(destination)
+            2 -> addedWaypoints.addSilent(destination)
+            else -> error("Not supported type of waypoint. index $index")
+        }
```

Comment:
ids and tags doesn't work for tabs. As this control isn't designed to use as a radio button I had to write this code 😃